crates/edr_provider/src/lib.rs

```diff
@@ -366,7 +402,17 @@ impl Provider {
             MethodInvocation::StopImpersonatingAccount(address) => {
                 hardhat::handle_stop_impersonating_account_request(data, *address).and_then(to_json)
             }
+        };
+
+        if let Some(method_name) = method_name {
+            // Only print errors for `hardhat_intervalMine`
+            if method_name != "hardhat_intervalMine" || result.is_err() {
```

Comment:
I might be misinterpreting the comment, but based on it, shouldn't it be `&&` instead of `||`?

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/4781/commits/1aced6e50b7a6af735bc476a550e56cd28eeab2d